Fix: Handle unescaped quotes in generate_value using safe_eval

milk333445 opened the pull request:
## Description
This PR improves the `generate_value` function in `NeMo-Guardrails/nemoguardrails/actions/llm/generation.py` to handle strings with unescaped single and double quotes more robustly. The updated implementation ensures that:
- Strings wrapped in quotes (single or double) are correctly evaluated using `literal_eval`.
- Strings without quotes are escaped and wrapped in valid Python string syntax before evaluation.
- The function gracefully handles errors caused by invalid input syntax.

Additionally, new test cases have been added `NeMo-Guardrails/tests/test_generate_value_safe_eval.py` to verify the correctness of `safe_eval` across different scenarios, including:
- Unquoted strings.
- Strings with nested single and double quotes.
- Empty strings.

All existing and new tests were successfully executed locally using `pytest` to ensure the changes do not introduce any regressions.



## Related Issue(s)

  - Fixes #938 


## Checklist

- [x] I've read the [CONTRIBUTING](https://github.com/NVIDIA/NeMo-Guardrails/blob/develop/CONTRIBUTING.md) guidelines.
- [x] I've updated the documentation if applicable.
- [x] I've added tests if applicable.
- [x] @mentions of the person or team responsible for reviewing proposed changes.

Pouyanpi commented on tests/test_generate_value_safe_eval.py:
Nice!

* FYR, only two of the tests fail with ast.literal_eval (but it is OK)
* Please move it to `test_utils.py` as you are testing the utility. And try to consolidate all the test-cases into one using `pytest.parametrize`.

Pouyanpi commented on nemoguardrails/actions/llm/generation.py:
```diff
@@ -1342,3 +1345,26 @@ def clean_utterance_content(utterance: str) -> str:
         # It should be translated to an actual \n character.
         utterance = utterance.replace("\\n", "\n")
     return utterance
+
```
please move it to `nemoguardrails/utils.py`

Pouyanpi reviewed (changes requested):
Thank you @milk333445 for your contribution 👍🏻  Please make the minor changes requested then we are good to merge

milk333445 commented:
The requested changes have been implemented as follows:
1. Moved the `safe_eval` function to `nemoguardrails/utils.py` as suggested.
2. Added logging to capture errors when exceptions are raised.
3. Refactored the test cases:
   - Moved the tests for `safe_eval` to `test_utils.py` as it is now part of the utilities.
   - Consolidated all test cases for `safe_eval` into a single function using `pytest.mark.parametrize`.

Pouyanpi commented:
@milk333445 thanks a lot, it was fast 👍🏻 Just make sure that you [sign](https://github.com/NVIDIA/NeMo-Guardrails/blob/f7db90ba323a34645d6dcfa01ed53fe9ac4dafca/CONTRIBUTING.md#submitting-your-work) your commits: you can do an interactive rebase and just sign them, then do a force push.

milk333445 commented:
I have signed all the commits, and they are now verified.
Please review the changes again and let me know if there's anything else needed. Thank you for your guidance!

Pouyanpi reviewed (approved):
LGTM! 👍🏻
